docs/zkapps/tutorials/oracle.mdx

```diff
@@ -0,0 +1,482 @@
+
+```ts
+// Get the oracle public key from the contract state
+const oraclePublicKey = this.oraclePublicKey.get();
+this.oraclePublicKey.assertEquals(oraclePublicKey);
+```
+
+We use `assertEquals()` to ensure that the public key we retrieved at execution time is the same as the public key that exists within the zkApp account on the Mina network when the transaction is processed by the network.
+
+### Verify the Signature
+
+Next, we’ll verify that the signature on the data (`id` and `creditScore`) is valid for the expected public key, to ensure it was from our expected source. This will return true if the signature is valid, and false if it is not.
+
+```ts
+// Evaluate whether the signature is valid for the provided data
+const validSignature = signature.verify(oraclePublicKey, [id, creditScore]);
+```
+
+We also want it to make it impossible to generate a valid zero-knowledge proof if `validSignature` is false. We can do this with `assertTrue()`. If the signature is invalid, this will throw an exception and make it impossible to generate a valid ZKP and transaction.
+
+```ts
+// Check that the signature is valid
+validSignature.assertTrue();
+```
+
+### Check that the Users Credit Score Is Above 700
+
+We want our `verify()` method to only emit an event if the user’s credit score is above 700. We can ensure that this condition is met by calling the `assertGte()` (i.e. assert greater than or equal to) on the `creditScore`.
+
+```ts
+// Check that the provided credit score is greater than 700
+creditScore.assertGte(Field(700));
+```
+
+These assert methods create a constraint that makes it impossible for users to generate a valid zero-knowledge proof unless their condition is met. Without a valid zero-knowledge proof (or a signature) it’s impossible to generate a valid Mina transaction. So, we can now rest assured that users can only call our smart contract method and send a valid transaction if they have a valid signature from our expected oracle and a credit score above 700.
+
+### Emitting our `verified` Event
+
+Now that we are sure everything checks out, we can emit an event to indicate this. The first argument to `emitEvent()` is an arbitrary string name chosen by the developer (because a smart contract could emit more than one type of event) and the second argument can be any value, as long as it matches the type defined for our event earlier. In this case, our event is `Field`, but it could be a more complicated type built on Fields, if the situation called for it. Emitted events are stored and available on archive nodes in the Mina network.
+
+```ts
+// Emit an event containing the verified users id
+this.emitEvent('verified', id);
+```
+
+## Testing it Out
+
+The zkApp CLI automatically generated a file called `CreditScoreOracle.test.ts` for us when we ran `zk file CreditScoreOracle`. Let’s add some tests. Open it and paste in the following:
```

Comment:
This should be `zk file CreditScoreVerifier` and `CreditScoreVerifier.test.ts` to match the `zk file` command from earlier.